request.go

```diff
@@ -146,12 +155,13 @@ func (e *booleanSyncResponse) Error() error {
 	return fmt.Errorf("API error: %s", e.DisplayText)
 }
 
-// AsyncInfo represents the details for any async call
-//
-// It retries at most Retries time and waits for Delay between each retry
-type AsyncInfo struct {
-	Retries int
-	Delay   int
+type asyncJob struct {
+	command      AsyncCommand
+	delay        int
+	retries      int
+	responseChan chan<- *AsyncJobResult
+	errorChan    chan<- error
+	ctx          context.Context
```

Comment:
Go style guide you mentioned a few days earlier: never embed a context into a struct.